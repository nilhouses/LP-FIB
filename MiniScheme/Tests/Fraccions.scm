
    ; TEST INTERACTIU!!! (llegeix de fitxer)
    
;Aquest test es pot executar des del directori arrel del projecte amb les comandes:

  ;SI VOLEM COMPROVAR EL CORRECTE FUNCIONAMENT DE LA GRAMÃ€TICA
    ;python3 scheme.py ./Tests/Fraccions.scm < ./Tests/Fraccions.inp | diff - ./Tests/Fraccions.out

  ;SI VOLEM VEURE LA SORTIDA PER PANTALLA
    ;python3 scheme.py ./Tests/Fraccions.scm < ./Tests/Fraccions.inp

  ;SI VOLEM ESCRIURE I LLEGIR PEL NOSTRE COMPTE
    ;python3 scheme.py ./Tests/Fraccions.scm


(define x (/ 3 2) )
(display "VARIABLE DEL CODI x = ")
(display x)
(newline)   
(display "x + 1/2 = ")
(display (+ x (/ 1 2))) ; 4/2 = 2
(newline)   
(display "x + 1 = ")
(display (+ x 1)) ; 5/2
(newline)

;Veiem que es puguin llegir fraccions
(define y (read))
(display "VARIABLE D'STDIN y = ")
(display y)
(newline)   
(display "y + 1/2 = ")
(display (+ y (/ 1 2))) ; 4/2 = 2
(newline)   
(display "y + 1 = ")
(display (+ y 1)) ; 5/2
(newline)