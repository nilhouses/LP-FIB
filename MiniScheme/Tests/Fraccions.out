VARIABLE DEL CODI x = 3/2
x + 1/2 = 2
x + 1 = 5/2
VARIABLE D'STDIN y = 3/2
y + 1/2 = 2
y + 1 = 5/2
